app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/texteditor/read/ReadTextFileTask.java

```diff
@@ -0,0 +1,172 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.amaze.filemanager.asynchronous.asynctasks.texteditor.read;
+
+import static com.amaze.filemanager.asynchronous.asynctasks.texteditor.read.ReadTextFileCallable.MAX_FILE_SIZE_CHARS;
+import static com.amaze.filemanager.filesystem.EditableFileAbstraction.Scheme.FILE;
+
+import java.io.IOException;
+import java.lang.ref.WeakReference;
+
+import com.amaze.filemanager.R;
+import com.amaze.filemanager.asynchronous.asynctasks.Task;
+import com.amaze.filemanager.file_operations.exceptions.StreamNotFoundException;
+import com.amaze.filemanager.ui.activities.texteditor.ReturnedValueOnReadFile;
+import com.amaze.filemanager.ui.activities.texteditor.TextEditorActivity;
+import com.amaze.filemanager.ui.activities.texteditor.TextEditorActivityViewModel;
+import com.google.android.material.snackbar.Snackbar;
+
+import android.content.Context;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.annotation.MainThread;
+import androidx.annotation.NonNull;
+import androidx.annotation.StringRes;
+import androidx.lifecycle.ViewModelProvider;
+
+public class ReadTextFileTask implements Task<ReturnedValueOnReadFile, ReadTextFileCallable> {
+  private static final String TAG = ReadTextFileTask.class.getSimpleName();
+
+  private final WeakReference<TextEditorActivity> textEditorActivityWR;
+  private final WeakReference<Context> appContextWR;
+  private final ReadTextFileCallable task;
+
+  public ReadTextFileTask(
+      TextEditorActivity activity,
+      WeakReference<TextEditorActivity> textEditorActivityWR,
+      WeakReference<Context> appContextWR) {
+    this.textEditorActivityWR = textEditorActivityWR;
+    this.appContextWR = appContextWR;
+
+    final TextEditorActivityViewModel viewModel =
+        new ViewModelProvider(activity).get(TextEditorActivityViewModel.class);
+
+    task =
+        new ReadTextFileCallable(
+            activity.getContentResolver(),
+            viewModel.getFile(),
+            activity.getExternalCacheDir(),
+            activity.isRootExplorer());
+  }
+
+  @NonNull
+  @Override
+  public ReadTextFileCallable getTask() {
+    return task;
+  }
+
+  @Override
+  @MainThread
+  public void onError(@NonNull Throwable error) {
+    Log.e(TAG, "Error on text read", error);
+
+    final Context applicationContext = appContextWR.get();
+    if (applicationContext == null) {
+      return;
+    }
+
+    @StringRes int errorMessage;
+
+    if (error instanceof StreamNotFoundException) {
```

Comment:
Can use `is` instead of `instanceof` in kotlin

Reply:
This is not in kotlin, should I convert it?